modify.go

```diff
@@ -155,3 +162,62 @@ func (l *Conn) Modify(modifyRequest *ModifyRequest) error {
 	}
 	return nil
 }
+
+// ModifyWithResult performs the ModifyRequest and returns the result
+func (l *Conn) ModifyWithResult(modifyRequest *ModifyRequest) (*ModifyResult, error) {
+	packet := ber.Encode(ber.ClassUniversal, ber.TypeConstructed, ber.TagSequence, nil, "LDAP Request")
+	packet.AppendChild(ber.NewInteger(ber.ClassUniversal, ber.TypePrimitive, ber.TagInteger, l.nextMessageID(), "MessageID"))
+	packet.AppendChild(modifyRequest.encode())
```

Comment:
Does this line compile for you?  There is no `encode()` method on the `ModifyRequest` type.

Reply:
@johnweldon: ah, there's not. i copy/pasta'd the code from when it was just modifying the original `Modify` method. I'll look into this and get it fixes.

Reply:
@johnweldon so looks like the `encode()` method was removed in https://github.com/go-ldap/ldap/pull/232/files#diff-25200df334664e32a437f35969897352L105. It's going to take me a bit to figure out what appears to be the replacement, `appendTo`.

It looks like it's a matter of replacing the encode stuff with `l.doRequest(...)`. Does that sound right?

Reply:
Ah, yes this changed in PR #232  - there are some overlapping changes so you may want to review that PR as you update yours.

Thanks again